core/src/main/java/org/jruby/Main.java

```diff
@@ -531,6 +537,26 @@ protected static int handleRaiseException(RaiseException rj) {
         }
     }
 
+    public static void printTruffleTimeMetric(String id) {
+        if (Options.TRUFFLE_METRICS_TIME.load()) {
+            final long millis = System.currentTimeMillis();
+            System.err.printf("%s %d.%03d%n", id, millis / 1000, millis % 1000);
+        }
+    }
+
+    private static void printTruffleMemoryMetric() {
+        if (Options.TRUFFLE_METRICS_MEMORY_USED_ON_EXIT.load()) {
+            System.gc();
```

Comment:
side note: there may be a reliable way how to force gc in JVMTI, http://docs.oracle.com/javase/7/docs/platform/jvmti/jvmti.html#ForceGarbageCollection. I did not use it though.